python/semantic_kernel/memory/volatile_memory_store.py

```diff
@@ -1,37 +1,300 @@
 # Copyright (c) Microsoft. All rights reserved.
 
+from copy import deepcopy
 from logging import Logger
-from typing import List, Optional, Tuple
+from typing import Dict, List, Optional, Tuple
 
 from numpy import array, linalg, ndarray
 
 from semantic_kernel.memory.memory_record import MemoryRecord
 from semantic_kernel.memory.memory_store_base import MemoryStoreBase
-from semantic_kernel.memory.storage.volatile_data_store import VolatileDataStore
 from semantic_kernel.utils.null_logger import NullLogger
 
 
-class VolatileMemoryStore(VolatileDataStore, MemoryStoreBase):
+class VolatileMemoryStore(MemoryStoreBase):
+    _store: Dict[str, Dict[str, MemoryRecord]]
+    _logger: Logger
+
     def __init__(self, logger: Optional[Logger] = None) -> None:
-        super().__init__()
+        self._store = {}
         self._logger = logger or NullLogger()
+        """Initializes a new instance of the VolatileMemoryStore class.
+
+        Arguments:
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+
+    async def create_collection_async(
+        self,
+        collection_name: str
+        # TODO: cancel token
```

Comment:
Trying to keep todos out of the code as much as possible. If you add these, can you please add an Issue to the github board, and add the task number to the TODO comment?

Reply:
removed TODOs for cancellation tokens and created a task to track the work